### ISSUE TITLE
fix: options knob not updating when switching from light to dark mode

### ISSUE THREAD

**Issue by lore-co** [widgetbook/widgetbook#557]
Widgetbook version: 2.4.1

I set up a Text use case Widget

```
  Widget build(BuildContext context) {
    var color = colorKnob(context);
    return Text(
        context.knobs
            .text(label: 'Text', initialValue: 'Lorem ipsum')
            .toString(),
        style: style?.copyWith(
        color: color,
        ),
      ),
    );
  }
```

I have added a knob to change colors from Widgetbook itself, where **WidgetBookOptions.themeData(context)** returns correctly a **List<Option<Color?>>** based on the current dark/light mode.

```
  Color? colorKnob(BuildContext context) {
    var options = WidgetBookOptions.themeData(context);
    return context.knobs.options(
      label: 'Text Color',
      options: options,
    );
  }
```

**var options** get correctly initialized with the new colors based on the current theme but the return value remains unchanged.

It only changes when I refresh the web page or I use hot reload.

P.S. Had a similar issue with the textstyles when I decided to move the Use Case in a separate widget, and solved it by passing a UniqueKey to it.

**Comment by jenshor** [comment]
@lore-co thanks. We'll look into it 👍🏻

**Comment by jenshor** [comment]
First of all, I'm able to reproduce your problem. Thanks for providing the repro code 👍🏻

Knobs work the following way: 
If a knob without a label does not exist, it'll be added to the knobs panel. If it already exists, adding it will be skipped and no value will be updated. 
Updates can only happen when the user triggeres changes via the user interface and cannot happen via code. 
I'll now have a look into the implementation and see if and how we can change the implementation to allow what you're requesting.

**Comment by lore-co** [comment]
> First of all, I'm able to reproduce your problem. Thanks for providing the repro code 👍🏻
> 
> Knobs work the following way: If a knob without a label does not exist, it'll be added to the knobs panel. If it already exists, adding it will be skipped and no value will be updated. Updates can only happen when the user triggers changes via the user interface and cannot happen via code. I'll now have a look into the implementation and see if and how we can change the implementation to allow what you're requesting.

That would be great! Let me know whether I can provide any assistance :)

**Comment by zoocityboy** [comment]
Maybe because  you have brightness for dark theme as Brightness.light?

**Comment by kekland** [comment]
@jenshor @lore-co I faced a similar issue with my custom knob to select colors from a predefined palette. What I ended up with is a knob that returns a selector function of type `(BuildContext) => Color`, and then this method is called in the `build` method of the story to resolve the color that is right for the context of the `build` method.

So the knob would look something like https://gist.github.com/kekland/d999f844e4d335ddf3057368b9b03f94.

This is not the most clean solution, but it worked fine on my end. Perhaps this might help with this issue.
